moveit_kinematics/test/test_kinematics_plugin.cpp

```diff
@@ -412,353 +390,358 @@ TEST_F(KinematicsTest, randomWalkIK)
     traj.addSuffixWayPoint(robot_state, 0.1);
   }
   EXPECT_LE(failures, (1.0 - EXPECTED_SUCCESS_RATE) * num_ik_tests_);
-  // TODO(JafarAbdi): Enable after porting the launch files
-  // if (publish_trajectory)
-  // {
-  //   ros::NodeHandle nh;
-  //   ros::AsyncSpinner spinner(1);
-  //   spinner.start();
-  //   ros::Publisher pub = nh.advertise<moveit_msgs::DisplayTrajectory>("display_random_walk", 1, true);
-  //   traj.getRobotTrajectoryMsg(msg.trajectory[0]);
-  //   pub.publish(msg);
-  //   ros::WallDuration(0.1).sleep();
-  // }
+  if (publish_trajectory_)
+  {
+    auto pub = node_->create_publisher<moveit_msgs::msg::DisplayTrajectory>("display_random_walk", 1);
+    traj.getRobotTrajectoryMsg(msg.trajectory[0]);
+    pub->publish(msg);
+    rclcpp::spin_some(node_);
+  }
+}
+
+static bool parsePose(const std::vector<double>& pose_values, Eigen::Isometry3d& goal)
+{
+  std::vector<double> vec;
+  Eigen::Quaterniond q;
+  if (pose_values.size() == 6)
+  {
+    q = Eigen::AngleAxisd(pose_values[3], Eigen::Vector3d::UnitX()) *
+        Eigen::AngleAxisd(pose_values[4], Eigen::Vector3d::UnitY()) *
+        Eigen::AngleAxisd(pose_values[5], Eigen::Vector3d::UnitZ());
+  }
+  else if (pose_values.size() == 7)
+  {
+    q = Eigen::Quaterniond(pose_values[3], pose_values[4], pose_values[5], pose_values[6]);
+  }
+  else
+  {
+    return false;
+  }
+
+  goal = q;
+  goal.translation() = Eigen::Vector3d(pose_values[0], pose_values[1], pose_values[2]);
+
+  return true;
+}
+
+TEST_F(KinematicsTest, unitIK)
+{
+  static const std::string TEST_POSES_PARAM = "unit_test_poses";
+  size_t expected_test_poses = 0;
+  node_->get_parameter_or(TEST_POSES_PARAM + ".size", expected_test_poses, expected_test_poses);
+
+  std::vector<double> sol;
+  const std::vector<std::string>& tip_frames = kinematics_solver_->getTipFrames();
+  moveit::core::RobotState robot_state(robot_model_);
+  robot_state.setToDefaultValues();
+  robot_state.setJointGroupPositions(jmg_, seed_);
```

Comment:
The original test checks if `seed_` is empty, is that case impossible now?

Reply:
We can check if it is empty or not, but every test declares a seed anyway.

Reply:
Could you add an assert that checks if `seed_` has the correct size?

Reply:
Sure thing

Reply:
There was already one: https://github.com/vatanaksoytezer/moveit2/blob/feature/enable-kinematic-tests/moveit_kinematics/test/test_kinematics_plugin.cpp#L117